### PR TITLE
Allow #[repr(align(x))] on enums (#57996)

### PR DESCRIPTION
**Pull request by niklasf**
Tracking issue: #57996

Implements an extension of [RFC 1358](https://github.com/rust-lang/rfcs/blob/master/text/1358-repr-align.md) behind a feature flag (`repr_align_enum`). Originally introduced here for structs: #39999.

It seems like only HIR-level changes are required, since enums are already aware of their alignment (due to alignment of their limbs).

cc @bitshifter

### CODE REVIEW

File: src/test/run-pass/structs-enums/align-enum.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+use std::mem;
+
+// Raising alignment
+#[repr(align(8))]
+enum Align8 {
+    Foo { foo: u32 },
+    Bar { bar: u32 },
+}
+
```
[the remark this answers is not in the document]
Comment (niklasf) [reply src/test/run-pass/structs-enums/align-enum.rs]: Thanks. Added.

**Comment by aturon** [comment]
We discussed this PR briefly in the Lang Team meeting today. Folks generally feel comfortable with this addition (especially since it's unstable). The one question was: is this equivalent to wrapping in a struct with an alignment? Modulo clarifying that part of the spec, we are :+1:

**Comment by nagisa** [comment]
This needs a codegen test IMO. Otherwise looks good.

**Comment by niklasf** [comment]
Makes sense. Added a very basic codegen test (a6fd6ec).

**Pull request merged**